compiler/compile/OMRSymbolReferenceTable.hpp

```diff
@@ -240,6 +240,29 @@ class SymbolReferenceTable
        */
       nonNullableArrayNullStoreCheckSymbol,
 
+      /**
+       * \brief This symbol represents a call that loads an array element
+       *
+       * \code
+       *   acall <loadFlattenableArrayElementNonHelperSymbol>
+       *     array-element-index
+       *     array-base-address
+       * \endcode
+       */
+      loadFlattenableArrayElementNonHelperSymbol,
+
+      /**
+       * \brief This symbol represents a call that stores a value into an array element
+       *
+       * \code
+       *   call <storeFlattenableArrayElementNonHelperSymbol>
+       *     value-reference-to-be-stored
+       *     array-element-index
+       *     array-reference-to-which-value-will-be-stored
+       * \endcode
+       */
+      storeFlattenableArrayElementNonHelperSymbol,
```

Comment:
>the array must have a reference type for its element type

I'm not too sure of what this statement means. Could you elaborate a bit more? Or give an example?

Reply:
I was just thinking about arrays of primitive types - `int[]` or `double[]` - these non-helpers (and the corresponding helpers) will never be used with those kinds of arrays.  It's not a big deal, but when someone encounters these helpers and non-helpers for the first time, it might avoid some confusion about with which kinds of arrays they are to be used.

Reply:
Addressed in 28d5327